### PR TITLE
[System.Net.Http] Fix CFNetworkHandler.CookieContainer not to return null

### PR DESCRIPTION
**Pull request by spouliot**
This aligns the handler with the managed HttpClientHandler [1] and make
it easier to switch between them without additional or conditional code.

[1] https://github.com/mono/mono/blob/master/mcs/class/System.Net.Http/System.Net.Http/HttpClientHandler.cs#L107

### CODE REVIEW

**Comment by akoeplinger** [comment]
Two small nits, LGTM otherwise:

1. don't link to `master` GitHub URLs, they'll be useless when the code moves around. Type `y` to get a permalink
2. the email in your commit seems wrong

File: mcs/class/System.Net.Http/CFNetworkHandler.cs [hunk 1]
```diff
@@ -100,7 +100,7 @@ void EnsureModifiability ()
 
 		public CookieContainer CookieContainer {
 			get {
-				return cookies;
+				return cookieContainer ?? (cookieContainer = new CookieContainer ());
```
Comment (alexrp) [inline mcs/class/System.Net.Http/CFNetworkHandler.cs]: Should be `cookies`, no?

Comment (marek-safar) [inline mcs/class/System.Net.Http/CFNetworkHandler.cs]: Yep this will fail as `cookieContainer` is undefined

Comment (spouliot) [inline mcs/class/System.Net.Http/CFNetworkHandler.cs]: yes, typo :( and we have something broken in System.Net.Http dependencies (XI build is special) so it did not break locally

**Comment by spouliot** [comment]
failures unrelated to commit

**Pull request merged**